stdlib/database/mysql/src/main/ballerina/mysql/Module.md

```diff
@@ -4,11 +4,95 @@ This module provides the functionality required to access and manipulate data st
 
 ### Endpoint 
 
-To access a database, you must first create a `client` object. Create a client of the MySQL Client type (i.e., `mysql:Client`) and provide the necessary connection parameters. This will create a pool of connections to the given MySQL database. A sample for creating a MySQL client can be found below.
+To access a database, you must first create a `client` object. A sample for creating a MySQL client can be
+found below.
+
+### Connection pool handling
+
+There are 3 possible scenarios for connection pool handling.
+
+1. Global, shareable default connection pool
+If you do not provide the `poolOptions` field, a globally shareable pool will be created for your database unless
+a connection pool matching with the properties you provided already exists.
+
+```ballerina
+mysql:Client testDB = new({
+    host: "localhost",
+    port: 3306,
+    name: "testdb",
+    username: "root",
+    password: "root",
+    dbOptions: { "useSSL": false }
+});
+```
+
+2. Client owned, unshareable connection pool
+If you define the `poolOptions` field inline, an unshareable connection pool will be created.
+
+```ballerina
+mysql:Client testDB = new({
+    host: "localhost",
+    port: 3306,
+    name: "testdb",
+    username: "root",
+    password: "root",
+    poolOptions: { maximumPoolSize: 5 },
+    dbOptions: { "useSSL": false }
+});
+```
+
+3. Local shareable connection pool
+If you create a record of type `sql:PoolOptions` and reuse that in the configuration of multiple clients, for each
+set of clients that connect to the same database instance with the same set of properties, a shared connection pool
+will be created.
```

Comment:
If you create a record of type `sql:PoolOptions` and reuse that in the configuration of multiple clients, a shared connection pool will be created for each set of clients that connect to the same database instance with the same set of properties.